api/src/main/java/io/grpc/RequestValidationInterceptor.java

```diff
@@ -0,0 +1,21 @@
+package io.grpc;
+
+public class RequestValidationInterceptor implements ServerInterceptor {
+
+    private RequestValidatorResolver requestValidatorResolver;
+
+    public RequestValidationInterceptor(RequestValidatorResolver requestValidatorResolver) {
+        this.requestValidatorResolver = requestValidatorResolver;
+    }
+
+    @Override
+    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+            ServerCall<ReqT, RespT> call,
+            Metadata headers,
+            ServerCallHandler<ReqT, RespT> next
+    ) {
+        ServerCall.Listener<ReqT> delegate = next.startCall(call, headers);
```

Comment:
`next` should be delayed as well, as from this point onward it isn't safe to call `call.close()` as call isn't thread-safe.

Reply:
Is delegate.close() safe instead of call.close() ?